src/createTRPCJotai.ts

```diff
@@ -69,17 +69,17 @@ const atomWithQuery = <TProcedure extends AnyQueryProcedure, TClient>(
   return queryAtom;
 };
 
-const atomWithMutation = <TProcedure extends AnyMutationProcedure, TClient>(
+const atomWithMutation = <TProcedure extends AnyTRPCMutationProcedure, TClient>(
   path: string[],
   getClient: (get: Getter) => TClient,
 ) => {
-  type Args = ProcedureArgs<TProcedure['_def']>;
+  type Args = inferProcedureInput<TProcedure>;
   type Output = inferProcedureOutput<TProcedure>;
   const mutationAtom = atom(
     null as Output | null,
-    async (get, set, args: Args) => {
+    async (get, set, [args]: [Args]) => {
```

Comment:
Is this change intentional? atoms can receive multiple arguments.

Reply:
Yes, this is intentional. If you use the released version with `trpc@11` and you attempt to send a string `test` argument, the `procedure.mutate(...args)` spreads that string and you get `procedure.mutate('t', 'e', 's', 't')`. The result of that is that I only get the first character sent over the network.

```
const appRouter = router({
  echo: publicProcedure.input(z.string()).query(({ input }) => {
    return "echo:" + input;
  }),
```

```
  const echoAtom = trpcClient.echo.atomWithQuery(() => "ping");
  console.log("echo", await store.get(echoAtom)); // echo p
```

I know atoms can have multiple args, but can trpc methods? Maybe it would be more clear to call this `Input` instead

Reply:
Oh, I misread it. I thought our original code is
```js
async (get, set, ...args: Args) => {
```
I don't remember why I didn't do so.